client/src/components/common/Search.jsx

```diff
@@ -0,0 +1,73 @@
+import React, { useEffect, useState } from 'react';
+import { AutoComplete } from 'antd';
+import { useHistory } from 'react-router-dom';
+import http from '../../services/httpService';
+import config from '../../services/config.json';
+import Img from './Img';
+
+import './css/search.css';
+
+const { Option } = AutoComplete;
+
+const SearchCity = () => {
+  const [value, setValue] = useState('');
+  const [expectedError, setExpectedError] = useState('لاتوجد مدن تبدأ بهذا الحرف');
+  const [result, setResult] = useState([]);
+
+  const { push } = useHistory();
+
+  const getCitiesName = async () => {
+    try {
+      const cities = await http.get(
+        `${config.apiEndPoint}/search?city=${value}`, { canelToken: http.source.token },
+      );
+      if (cities.data.data.length) {
+        setResult(cities.data.data);
+      }
+    } catch (error) {
+      setExpectedError(error.message);
```

Comment:
error here could be 400 error, not for empty result error.
So maybe alert would be better soultion